pandas/core/frame.py

```diff
@@ -7973,7 +7973,7 @@ def corr(self, method="pearson", min_periods=1) -> "DataFrame":
         numeric_df = self._get_numeric_data()
         cols = numeric_df.columns
         idx = cols.copy()
-        mat = numeric_df.astype(float, copy=False).to_numpy()
+        mat = numeric_df.to_numpy(dtype=float, na_value=np.nan)
```

Comment:
do we have copy=False by default in `.to_numpy()`?

Reply:
Yes but I can make this explicit